### PR TITLE
TW-285: add loader

### PR DESCRIPTION
**Pull request by letier3110**

### CODE REVIEW

File: src/components/loader/loader-lines.tsx [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+          strokeWidth="1.5"
+          strokeLinecap="round"
+          d="M0,32a32,16 0 1,0 64,0a32,16 0 1,0 -64,0Z"
+        />
+      </G>
+    </Svg>
+  );
+};
+
+const useAnimatedGradient = (progress: SharedValue<number>, offset = -Math.PI / 2) => {
```
Comment (LendiHop) [inline src/components/loader/loader-lines.tsx:70]: short return

File: src/screens/enter-password/enter-password.tsx [hunk 2]
```diff
@@ -47,7 +50,16 @@ export const EnterPassword = () => {
...
+  const onSubmit = ({ password }: EnterPasswordFormValues) => {
+    if (!isDisabled) {
+      dispatch(setLoadingAction(true));
+      unlock(password);
+    }
+  };
+
+  useEffect(() => {
+    dispatch(setLoadingAction(false));
+  }, []);
```
Comment (LendiHop) [inline src/screens/enter-password/enter-password.tsx:62]: why do you need this here?

Comment (letier3110) [reply src/screens/enter-password/enter-password.tsx:62]: at least one point in application to prevent freezed loading

**Review by LendiHop — DISMISSED** [review]

File: src/shelter/utils/reveal-secrets-subscription.util.ts [hunk 3]
```diff
@@ -1,27 +1,38 @@
...
           .pipe(map((seedPhrase): [string | undefined, EventFn<string>] => [seedPhrase, successCallback]))
-      )
+      ),
+      tap(() => dispatch(setLoadingAction(false)))
     )
-  ).subscribe(([value, successCallback]) => value !== undefined && successCallback(value));
+  ).subscribe(([value, successCallback]) => {
+    if (value !== undefined) {
+      successCallback(value);
+    }
```
Comment (unixvb) [inline src/shelter/utils/reveal-secrets-subscription.util.ts:37]: simplify as it was before

**Review by unixvb — APPROVED** [review]

File: src/modals/import-account-modal/import-account-seed/import-account-seed.tsx [hunk 4]
```diff
@@ -41,13 +41,15 @@ export const ImportAccountSeed: FC<Props> = ({ onBackHandler }) => {
...
 
-    createImportedAccount({
-      name: `Account ${accountsIndex}`,
-      privateKey
-    });
+      createImportedAccount({
+        name: `Account ${accountsIndex}`,
+        privateKey
+      });
+    }, 100);
```
[the remark this answers is not in the document]
Comment (letier3110) [reply src/modals/import-account-modal/import-account-seed/import-account-seed.tsx:52]: the `mnemonicToSeedSync` is very heavy synchronous operation, that causes UI to freeze even on high-end emulators. So it basically blocks execution of other UI elements (like showing loader) until its done

File: src/shelter/use-app-lock.hook.ts [hunk 5]
```diff
@@ -35,13 +36,21 @@ export const useAppLock = () => {
 
   useEffect(() => {
     const subscriptions = [
-      Shelter.isLocked$.subscribe(value => setIsLocked(value)),
+      Shelter.isLocked$.subscribe(value => {
+        dispatch(setLoadingAction(false));
+        if (value === false) {
+          dispatch(enterPassword.success());
+        }
```
Comment (unixvb) [inline src/shelter/use-app-lock.hook.ts:43]: why this code appears?  unlock was working good

File: src/modals/import-account-modal/import-account-seed/import-account-seed.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (unixvb) [reply src/modals/import-account-modal/import-account-seed/import-account-seed.tsx:52]: set settimeout 0 and make mechanism against loader flickering

**Review by NovakVlad2424 — APPROVED** [review]

**Pull request merged**